java/memory/src/main/java/org/apache/arrow/memory/ArrowBuf.java

```diff
@@ -872,6 +874,7 @@ public void setBytes(long index, ByteBuffer src) {
           --length;
           ++dstAddress;
         }
+        src.order(originalByteOrder);
```

Comment:
should we place this in a finally block?

Reply:
Good idea, I've updated the PR.